src/nvim/window.c

```diff
@@ -5285,6 +5285,25 @@ static void frame_remove(frame_T *frp)
   }
 }
 
+/// Check if the new Nvim application "screen" dimensions are valid.
+/// Correct it if it's too small or way too big.
+void check_screensize(void)
```

Comment:
Is this really the best place for this?

Reply:
In Vim this function is called `limit_screen_size()`, and it is next to `win_new_shellsize()` in `term.c`. In Nvim `win_new_shellsize()` has been moved to `win_new_screensize()` in `window.c`, so I think it makes sense to also move `check_screensize()` here.

Reply:
I think even so, the contexts of both are too different to try and be consistent with each other. I would move this back to either `screen.c` or `drawscreen.c`.

Reply:
I guess `screen.c` is better for this function then.